tests/cnf/core/network/metallb/tests/metallb-nodeSelector.go

```diff
@@ -0,0 +1,300 @@
+				Expect(len(output.Routes)).To(Equal(2))
+
+				By("Validate BGP Custom Community 500:500 exists with the node selector")
+				_, err = frr.GetBGPCommunityStatus(frrPod1, tsparams.CustomCommunity, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
+				Expect(len(output.Routes)).To(Equal(2))
+			})
+	})
+
+	Context("Dual IPAddressPools", func() {
+
+		var addressPool2 = []string{"4.4.4.1", "4.4.4.240"}
+
+		BeforeAll(func() {
+
+			By("Creating a new instance of MetalLB Speakers on workers")
+			err = metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+			Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+			waitForWebHookRunning()
+		})
+
+		AfterEach(func() {
+			By("Clean metallb operator and test namespaces")
+			resetOperatorAndTestNS()
+		})
+
+		It("Advertise separate IPAddressPools using the node selector",
+			reportxml.ID("53986"), func() {
+				By("Create two IPAddressPools")
+				ipAddressPool1 := createIPAddressPool(ipaddressPoolName1, addressPool)
+				ipAddressPool2 := createIPAddressPool(ipaddressPoolName2, addressPool2)
+
+				frrPod0, frrPod1 := setupTestCase(ipAddressPool1, ipAddressPool2, frrk8sPods)
+
+				By("Creating a BGPAdvertisement with nodeSelector for bgpPeer1")
+				worker0NodeLabel := []metav1.LabelSelector{
+					{MatchLabels: map[string]string{netparam.LabelHostName: workerNodeList[0].Definition.Name}},
+				}
+
+				worker1NodeLabel := []metav1.LabelSelector{
+					{MatchLabels: map[string]string{netparam.LabelHostName: workerNodeList[1].Definition.Name}},
+				}
+
+				By("Creating a BGPAdvertisement with the nodeSelector to bgppeer1")
+				setupBgpAdvertisement(bgpAdvertisementName1, tsparams.NoAdvertise, ipaddressPoolName1,
+					100, []string{tsparams.BgpPeerName1}, worker0NodeLabel)
+
+				setupBgpAdvertisement(bgpAdvertisementName2, tsparams.CustomCommunity, ipaddressPoolName2,
+					200, []string{tsparams.BgpPeerName2}, worker1NodeLabel)
+
+				verifyBGPConnectivityAndPrefixes(frrPod0, frrPod1, nodeAddrList, addressPool, addressPool2)
+
+				By("Validate Local Preference from Frr node0")
+				err = frr.ValidateLocalPref(frrPod0, 100, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Fail to validate local preference")
+
+				By("Validate Local Preference from Frr node1")
+				err = frr.ValidateLocalPref(frrPod1, 200, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Fail to validate local preference")
+
+				By("Validate BGP Community exists with the node selector")
+				output, err := frr.GetBGPCommunityStatus(frrPod0, tsparams.NoAdvertise, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
+				Expect(len(output.Routes)).To(Equal(1))
+
+				By("Validate BGP Community does not exist without the node selector")
```

Comment:
How do we check that the comunity doesn't exist?

Reply:
I changed the description. We can add a negitive check as well. But not sure it is needed. We are just validating the the correct community has been added to the route.